Tools/ardupilotwaf/boards.py

```diff
@@ -774,22 +774,31 @@ def configure_env(self, cfg, env):
         # whitelist of compilers which we should build with -Werror
         gcc_whitelist = frozenset([
                 ('11','3','0'),
+                ('12','1','0'),
```

Comment:
[Ubuntu 23 has 12.2.0](https://packages.ubuntu.com/lunar/gcc) did you mean to add that instead?

Reply:
If you add this, the build will fail, because of #24142